samples/RewriteSample/Startup.cs

```diff
@@ -19,13 +20,61 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
                 .AddRewrite(@"app/(\d+)", "app?id=$1", skipRemainingRules: false)
                 .AddRedirectToHttps(302, 5001)
                 .AddIISUrlRewrite(env.ContentRootFileProvider, "UrlRewrite.xml")
-                .AddApacheModRewrite(env.ContentRootFileProvider, "Rewrite.txt");
+                .AddApacheModRewrite(env.ContentRootFileProvider, "Rewrite.txt")
+                .AddIISUrlRewrite(CreateCustomIISRule());
 
             app.UseRewriter(options);
-
             app.Run(context => context.Response.WriteAsync($"Rewritten Url: {context.Request.Path + context.Request.QueryString}"));
         }
 
+        /// <summary>
+        /// Creates an IIS rewrite rule programmatically that is equivalent to the following xml configuration:
+        /// 
+        /// <rule name="Test">
+        ///     <match url="(.*)" ignoreCase="false" />
+        ///     <conditions trackAllCaptures = "true" >
+        ///         <add input="{REQUEST_URI}" pattern="^/([a-zA-Z]+)/([0-9]+)$" />
+        ///         <add input="{QUERY_STRING}" pattern="p2=([a-z]+)" />
+        ///     </conditions>
+        ///     <action type="Redirect" url ="blogposts/{C:1}/{C:4}" />
+        /// </rule>
+        /// </summary>
+        /// <returns>An <see cref="IISUrlRewriteRule"/></returns>
+        private static IISUrlRewriteRule CreateCustomIISRule()
```

Comment:
That's unfortunate. I specifically need to build some rules from an external configuration (e.g., from a service or stored in a database). I would imagine this isn't that uncommon of a scenario. But, ok.